JitsiConference.js

```diff
@@ -1071,60 +1083,52 @@ JitsiConference.prototype.getTranscriptionStatus = function() {
+        const localTracks = this.rtc.getLocalTracks(mediaType);
+        const videoType = track.getVideoType();
 
-        // Currently, only adding multiple video streams of different video types is supported.
-        // TODO - remove this limitation once issues with jitsi-meet trying to add multiple camera streams is fixed.
-        if (mediaType === MediaType.VIDEO
-            && !localTracks.find(t => t.getVideoType() === track.getVideoType())) {
-            const sourceName = getSourceNameForJitsiTrack(
-                this.myUserId(),
-                mediaType,
-                this.getLocalTracks(mediaType)?.length);
-
-            track.setSourceName(sourceName);
-            const addTrackPromises = [];
-
-            this.p2pJingleSession && addTrackPromises.push(this.p2pJingleSession.addTracks([ track ]));
-            this.jvbJingleSession && addTrackPromises.push(this.jvbJingleSession.addTracks([ track ]));
-
-            return Promise.all(addTrackPromises)
-                .then(() => {
-                    this._setupNewTrack(track);
-                    this._sendBridgeVideoTypeMessage(track);
-                    this._updateRoomPresence(this.getActiveMediaSession());
-
-                    if (this.isMutedByFocus || this.isVideoMutedByFocus) {
-                        this._fireMuteChangeEvent(track);
-                    }
-                });
+        if (localTracks.find(t => t === track)) {
+            // a NOOP if the track is in the conference already
+            return;
+        } else if (track.isVideoTrack()
+            && this.getLocalVideoTracks().find(t => t.getVideoType() === track.getVideoType())) {
+            // Currently, only adding multiple video streams of different video types is supported
+            // TODO remove this limitation once issues with jitsi-meet trying to add multiple camera streams
+            //      is fixed.
+            throw new Error(`Cannot add second "${videoType}" video track`);
         }
 
-        return Promise.reject(new Error(`Cannot add second ${mediaType} track to the conference`));
-    }
+        const sourceName = getSourceNameForJitsiTrack(
+            this.myUserId(),
+            mediaType,
+            this.getLocalTracks(mediaType)?.length);
 
-    return this.replaceTrack(null, track)
-        .then(() => {
-            // Presence needs to be sent here for desktop track since we need the presence to reach the remote peer
-            // before signaling so that a fake participant tile is created for screenshare. Otherwise, presence will
-            // only be sent after a session-accept or source-add is ack'ed.
-            if (track.getVideoType() === VideoType.DESKTOP) {
+        track.setSourceName(sourceName);
+        const addTrackPromises = [];
+
+        this.p2pJingleSession && addTrackPromises.push(this.p2pJingleSession.addTracks([ track ]));
+        this.jvbJingleSession && addTrackPromises.push(this.jvbJingleSession.addTracks([ track ]));
```

Comment:
As discussed, this is going to be problematic. In both jvb and p2p case, the first track needs to be added to be replaced on the `RTCRtpSender` associated with the first `recvonly` transceiver. When a secondary track is added, for jvb, a new transceiver is created and for p2p, a new transceiver is created only when there is no `recvonly` one. JingleSessionPC::addTracks needs to be modified to take this into account. Right now, it is designed to be used for secondary tracks only and therefore creates a new transceiver for jvb always.

Reply:
By primary and secondary do you mean for each media type, for example:
- primary track will be the 1st video track added to the connection
- secondary track will be the 2nd video track added to the connection?

If there are two tracks, one audio and one video then there is one primary video track and one primary audio track?